README.md

```diff
@@ -306,6 +307,33 @@ Given a value `m`, one can access its type representative via the
 
 1. `empty` must return a value of the same Monoid
 
+### Filterable
+
+A value that implements the Filterable specification must also implement
+the [Monoid](#monoid) specification.
```

Comment:
For the reason given in sanctuary-js/sanctuary-type-classes#37 I think Filterable should depend on Plus rather than Monoid. `filter (odd) (Just (42))` should be valid.